Cloudbase init switch

ilenghel opened the pull request:
Added a switch parameter to select which Cloudbase-Init branch to use.

ader1990 committed:
Added Golden Image support

ilenghel committed:
Add support for VirtioDriver on golden images

Added Get-ImageInformation method for image information introspection
and option for virtio drivers application on the golden images

andreitira committed:
Added option to create password protected zip

ader1990 committed:
Added recursive golden image option and fixes

ader1990 committed:
Reboot after every update

costingalan commented on WinImageBuilder.psm1:
```diff
@@ -351,14 +351,25 @@ function Download-CloudbaseInit {
-        [string]$osArch
+        [string]$osArch,
+        [parameter(Mandatory=$false)]
+        [switch]$develBranch=$true
```
I would change from `$develBranch` to something like `$betaRelease`

costingalan replied:
I would change `$develBranch` to something like `$betaRelease`

costingalan commented on WinImageBuilder.psm1:
```diff
@@ -351,14 +351,25 @@ function Download-CloudbaseInit {
+        [switch]$develBranch=$true
+
     )
     Write-Host "Downloading Cloudbase-Init..."
```
I would also believe a better way to address the Cloudbase-Init installation is to write a function for this.

costingalan commented:
Also please do a `git rebase` in order to have only one commit in the end.

ilenghel committed:
Added Cloudbase-Init-Switch parameter

Cloudbase-Init has two branches, stable and beta. By setting the
cloudbaseInitDevel parameter to false, the stable version will be
downloaded and installed, otherwise the beta branch will be used.

ilenghel commented:
Wrong branch submitted. Closing PR.
